projects/js-toolkit/packages/portal-base/src/build/themeSpritemap.ts

```diff
@@ -0,0 +1,172 @@
+		abort(`Source directory, (${project.srcDir.asNative}), doesn't exist.`);
+	}
+
+	let svgContent =
+		'<?xml version="1.0" encoding="UTF-8"?>' +
+		'<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">' +
+		'<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">';
+
+	if (extendClay) {
+		const clayPath = require.resolve('@clayui/css');
+
+		if (!clayPath) {
+			print(
+				error` @clay/css package not found. Add as a dependency to your client extension via npm.`
+			);
+		}
+
+		const claySpritemapPath = require.resolve(
+			'@clayui/css/lib/images/icons/icons.svg'
+		);
+
+		const claySpritemapContent = fs.readFileSync(claySpritemapPath, 'utf8');
+
+		svgContent = claySpritemapContent
+			.replace('</svg>', '')
+			.replace(/\n/gm, '')
+			.replace(/\t/gm, '');
+	}
+
+	const filePaths = globby
+		.sync(['**/*.svg'], {
+			cwd: project.srcDir.asNative,
+		})
+		.map((posixPath) => new FilePath(posixPath, {posix: true}))
+		.map((file) => path.join(project.srcDir.asNative, file.asNative));
+
+	if (!filePaths.length) {
+		abort(`No icon(.svg) files found.`);
+	}
+
+	const iconsReplaced = [];
+
+	for (const filePath of filePaths) {
+		const fileName = path.basename(filePath, '.svg').toLowerCase();
+
+		const data = fs.readFileSync(filePath, 'utf8');
+
+		// Remove existing Clay icons that duplicate our new icon names
+
+		if (extendClay) {
+			const existingSymbolRegex = new RegExp(
+				`<symbol id="${fileName}".*?</symbol>`,
+				'gm'
+			);
+
+			if (existingSymbolRegex.test(svgContent)) {
+				svgContent = svgContent.replace(existingSymbolRegex, '');
+
+				iconsReplaced.push(fileName);
+			}
+		}
+
+		svgContent += data
+			.replace(HEADER_REGEXP, '')
+			.replace(/<svg/gm, '<symbol')
+			.replace(
+				/xmlns="http:\/\/www\.w3\.org\/2000\/svg"/gm,
+				`id="${fileName}"`
+			)
+			.replace(/<\/svg/gm, '</symbol')
+			.replace(/\n/gm, '')
+			.replace(/\t/gm, '');
+	}
+
+	const iconsAdded = filePaths.length - iconsReplaced.length;
+
+	if (iconsAdded) {
+		print(info` ${iconsAdded} new icons added.`);
+	}
+
+	if (extendClay) {
+		print(
+			info` Extended Clay spritemap and replaced ${
+				iconsReplaced.length
+			} icons: ${iconsReplaced.join(', ')}.`
+		);
+	}
+
+	svgContent += '</svg>';
+
+	fs.writeFileSync(
+		path.join(project.build.dir.asNative, SPRITEMAP_FILE_NAME),
+		svgContent
+	);
+
+	const configurationPid =
+		'com.liferay.client.extension.type.configuration.CETConfiguration~' +
+		project.pkgJson.name;
+
+	const clientExtensionConfigJson: ClientExtensionConfigJson = {
+		[configurationPid]: {
+			baseURL: `\${portalURL}/o/${project.pkgJson.name}`,
+			description: project.pkgJson.description || '',
+			name: project.pkgJson.name,
+			sourceCodeURL: '',
+			type: 'themeSpritemap',
+			typeSettings: [
+				`enableSVG4Everybody=${enableSVG4Everybody}`,
+				`url=${SPRITEMAP_FILE_NAME}`,
+			],
+		},
+	};
+
+	const zip = new JSZip();
+
+	zip.file(
```

Comment:
We may need to refactor all this stuff in the future bc we surely don't want to have the (almost) same zipping code in every CE type.